### PR TITLE
Coexpression: only send common samples in api request

### PR DESCRIPTION
**Pull request by kalletlak**
Fix https://github.com/cBioPortal/cbioportal/issues/7403

### CODE REVIEW

**Comment by alisman** [comment]
@n1zea144 can one of your team members look at this one?

File: src/pages/resultsView/coExpression/CoExpressionTab.tsx [hunk 1]
```diff
@@ -261,16 +259,46 @@ export default class CoExpressionTab extends React.Component<
...
+                const profileXSamples = this.props.store
+                    .molecularProfileIdToProfiledSamples.result![
+                    q.profileX.molecularProfileId
+                ];
+                const profileYSamples = this.props.store
+                    .molecularProfileIdToProfiledSamples.result![
+                    q.profileY.molecularProfileId
+                ];
+
+                let dataFilter: IDataQueryFilter = {} as any;
```
Comment (Luke-Sikina) [inline src/pages/resultsView/coExpression/CoExpressionTab.tsx:281]: Don't change this unless there are other changes to make too, but you don't have to cast `{}` to `any`, as `IDataQueryFilter` only has optional fields.

File: src/pages/resultsView/coExpression/CoExpressionTab.tsx [hunk 2]
```diff
@@ -261,16 +259,46 @@ export default class CoExpressionTab extends React.Component<
...
+                if (
+                    q.profileX.molecularProfileId ===
+                    q.profileY.molecularProfileId
+                ) {
+                    dataFilter = this.props.store.studyToDataQueryFilter
+                        .result![q.profileX.studyId];
+                } else {
+                    const samples = _.intersectionWith(
```
Comment (alisman) [inline src/pages/resultsView/coExpression/CoExpressionTab.tsx:289]: commonSamples

**Comment by kalletlak** [comment]
Failing e2e localdb tests is related to backend api bug( fixed in https://github.com/cBioPortal/cbioportal/pull/7429)

**Pull request merged**